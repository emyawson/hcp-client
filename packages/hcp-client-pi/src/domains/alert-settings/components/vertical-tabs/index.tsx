export {
  VerticalTab,
  VerticalTabs,
  VerticalTabsProps,
} from './vertical-tabs.component';
