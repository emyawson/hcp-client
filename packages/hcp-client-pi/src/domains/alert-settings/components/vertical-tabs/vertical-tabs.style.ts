import { IThemeInterface } from '@roche/patterns-indicators/theme';
import * as React from 'react';
import styled, { StyledComponentClass } from 'styled-components';

/* TODO: move border styles outside of this base component and apply to a wrapper component */
export const Container: StyledComponentClass<null, IThemeInterface> = styled<
  null,
  'div'
>('div')`
  display: flex;
  padding: ${({ theme }) => theme.spacing.four};
  height: 100%;
  width: 100%;
  background-color: ${props => props.theme.colors.white};
  border-left: 0.0625rem solid ${props => props.theme.colors.silverDark};
  border-bottom: 0.0625rem solid ${props => props.theme.colors.silverDark};
  border-right: 0.0625rem solid ${props => props.theme.colors.silverDark};
`;

export const Nav: StyledComponentClass<
  { onChange?: (event: React.ChangeEvent<any>) => void },
  IThemeInterface
> = styled.nav`
  padding-right: 3rem;
  width: 24rem;
  min-width: 24rem;
`;

export const NavTabsList: StyledComponentClass<{}, IThemeInterface> = styled.ul`
  list-style-type: none;
  height: 100%;
  padding: 0;
  margin: 0;
`;

interface NavTabsListItemProps {
  selected: boolean;
  selectedStyle: string;
  onClick?: () => void;
}

export const NavTabsListItem: StyledComponentClass<
  NavTabsListItemProps,
  IThemeInterface
> = styled<NavTabsListItemProps, 'li'>('li')`
  cursor: pointer;
  border-bottom: 0.0625rem solid ${({ theme }) => theme.colors.grayLight};
  display: flex;
  background-color: ${({ selected, theme }) =>
    selected ? theme.colors.silverLight : theme.colors.white};
  font-weight: ${({ theme }) => theme.fontWeights.semiBold};
  line-height: 2;
  font-size: ${({ theme }) => theme.fontSizes.label};
`;
NavTabsListItem.displayName = 'NavTabsListItem';

export const NavTabsListTitle: StyledComponentClass<
  {},
  IThemeInterface
> = styled.div`
  margin-bottom: 4rem;
`;

interface SelectedLineProps {
  show: boolean;
}
export const SelectedLine: StyledComponentClass<
  SelectedLineProps,
  IThemeInterface
> = styled<SelectedLineProps, 'span'>('span')`
  display: inline-block;
  min-height: 4rem;
  position: relative;
  border-left: ${({ show, theme }) =>
    show ? `3px solid ${theme.colors.blue}` : `3px solid transparent`};
  border-radius: ${({ theme }) => theme.borderRadius[2]};
`;

export const NavTabsListItemText: StyledComponentClass<
  {},
  IThemeInterface
> = styled.div`
  display: inline-block;
  padding: ${({ theme }) => theme.spacing.two};
  margin: auto 0;
  width: 100%;
`;

export const ContentContainer: StyledComponentClass<
  {},
  IThemeInterface
> = styled.section`
  margin-left: ${({ theme }) => theme.spacing.two};
  padding: 0 ${({ theme }) => theme.spacing.one};
  width: 100%;
`;
