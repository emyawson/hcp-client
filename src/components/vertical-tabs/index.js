export * from './vertical-tabs.component';
