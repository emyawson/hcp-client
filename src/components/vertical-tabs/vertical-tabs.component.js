import React from 'react';

import {
  Container,
  ContentContainer,
  Nav,
  NavTabsList,
  NavTabsListItem,
  NavTabsListItemText,
  SelectedLine,
} from './vertical-tabs.style';

export class VerticalTabs extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedTabIndex: 0,
    };
  }

  render() {
    const {
      height,
      minHeight,
      navItemLabelSelectedStyle,
      tabs,
      m,
    } = this.props;
    return (
      <Container minHeight={minHeight} m={m}>
        <Nav>
          <NavTabsList>
            {tabs.map((tab, index) => (
              <NavTabsListItem
                key={tab.keyText}
                onClick={() => this.setState({ selectedTabIndex: index })}
                selected={index === this.state.selectedTabIndex}
                selectedStyle={navItemLabelSelectedStyle}
              >
                <SelectedLine show={index === this.state.selectedTabIndex} />
                <NavTabsListItemText
                  selected={index === this.state.selectedTabIndex}
                >
                  {tab.label}
                </NavTabsListItemText>
              </NavTabsListItem>
            ))}
          </NavTabsList>
        </Nav>
        <ContentContainer height={height}>
          {tabs[this.state.selectedTabIndex].content}
        </ContentContainer>
      </Container>
    );
  }
}
