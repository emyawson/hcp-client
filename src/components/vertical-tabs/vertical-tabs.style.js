import styled from 'styled-components';

import { borderRadius, colors, spacing } from 'src/core';
import { convertPxToRem } from 'src/utils';

export const Container = styled.div`
  flex: 1;
  display: flex;
  border-top: 0.0625rem solid ${colors.grayLight};
  height: 100%;
  height: ${props => props.minHeight};
  min-height: ${props => props.minHeight};
`;

export const Nav = styled.nav`
  flex-shrink: 1;
`;

export const NavTabsList = styled.ul`
  list-style-type: none;
  border-right: 0.0625rem solid ${colors.grayLight};
  height: 100%;
  padding: 0;
  margin: 0;
  width: 2.5rem;
`;

export const NavTabsListItem = styled.li`
  cursor: pointer;
  border-bottom: 0.0625rem solid ${colors.grayLight};
  margin-right: ${props => (props.selected ? '-0.0625rem' : '0')};
  border-right: ${props =>
    props.selected ? `0.0625rem solid ${colors.white}` : 'none'};
  ${props =>
    props.selected && props.selectedStyle ? props.selectedStyle : null};
  &:hover {
    ${props => (props.selectedStyle ? props.selectedStyle : null)};
  }

  height: 2.5rem;
`;
NavTabsListItem.displayName = 'NavTabsListItem';

export const SelectedLine = styled.div`
  display: inline-block;
  height: 1.5rem;
  position: relative;
  top: ${spacing.two};
  border-left: ${props =>
    props.show
      ? `${convertPxToRem(3)} solid ${colors.blue}`
      : `${convertPxToRem(3)} solid transparent`};
  border-radius: ${borderRadius.three};
`;

export const NavTabsListItemText = styled.div`
  display: inline-block;
  padding: ${spacing.one} ${spacing.two};
`;

export const ContentContainer = styled.section`
  flex: 1;
  padding: 0 ${spacing.one};
`;
