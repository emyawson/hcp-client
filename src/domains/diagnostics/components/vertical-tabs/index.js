export { SelectedLine } from './vertical-tabs.style';
export { VerticalTabs } from './vertical-tabs.component';
