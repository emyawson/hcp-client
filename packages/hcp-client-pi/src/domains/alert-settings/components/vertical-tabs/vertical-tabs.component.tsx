import * as React from 'react';
import { withTheme } from 'styled-components';

import { SectionHeader } from '@roche/patterns-indicators/components';
import { RenderIf } from '@roche/patterns-indicators/utils/markup/render-if.utils';

import { Tab } from '../../alert-settings.types';

import {
  Container,
  ContentContainer,
  Nav,
  NavTabsList,
  NavTabsListItem,
  NavTabsListItemText,
  NavTabsListTitle,
  SelectedLine,
} from './vertical-tabs.style';

export interface VerticalTab extends Tab {
  label: JSX.Element;
  toggled: boolean;
  selected: boolean;
}

export interface VerticalTabsProps {
  headerText?: string;
  minHeight?: number;
  navItemLabelSelectedStyle?: string;
  setSelectedTabIndex: (index: number) => void;
  tabs?: VerticalTab[];
  theme: any;
}

interface VerticalTabsState {
  selectedTabIndex: number;
}
class VerticalTabsComponent extends React.Component<
  VerticalTabsProps,
  VerticalTabsState
> {
  constructor(props: VerticalTabsProps) {
    super(props);
    this.state = {
      selectedTabIndex: 0,
    };
  }

  public render() {
    const {
      headerText,
      navItemLabelSelectedStyle = '',
      tabs = [],
      theme,
    } = this.props;
    return (
      <RenderIf validate={tabs.length}>
        <Container>
          <Nav onChange={this.tabsChangeHandler}>
            <RenderIf validate={headerText}>
              <NavTabsListTitle>
                <SectionHeader
                  borderColor={theme.colors.silverDark}
                  title={headerText}
                />
              </NavTabsListTitle>
            </RenderIf>
            <NavTabsList>
              {tabs.map((tab, index) => (
                <NavTabsListItem
                  key={`tab ${index}`}
                  onClick={
                    tab.enabled ? this.setSelectedTabIndex(index) : undefined
                  }
                  selected={index === this.state.selectedTabIndex}
                  selectedStyle={navItemLabelSelectedStyle}
                >
                  <SelectedLine show={index === this.state.selectedTabIndex} />
                  <NavTabsListItemText>{tab.label}</NavTabsListItemText>
                </NavTabsListItem>
              ))}
            </NavTabsList>
          </Nav>
          <ContentContainer>
            {!tabs[this.state.selectedTabIndex]
              ? null
              : tabs[this.state.selectedTabIndex].content}
          </ContentContainer>
        </Container>
      </RenderIf>
    );
  }

  private setSelectedTabIndex = (index: number) => () => {
    this.props.setSelectedTabIndex(index);
    this.setState({ selectedTabIndex: index });
  };

  private tabsChangeHandler = (event: React.ChangeEvent<any>) => {
    const tabIndex = event.target.id;
    const isChecked = event.target.checked;
    const elementType = event.target.type;

    if (elementType === 'checkbox') {
      if (!isChecked) {
        this.setSelectedTabIndex(0)();
      } else {
        this.setSelectedTabIndex(Number(tabIndex))();
      }
    }
  };
}

export const VerticalTabs = withTheme(VerticalTabsComponent);
