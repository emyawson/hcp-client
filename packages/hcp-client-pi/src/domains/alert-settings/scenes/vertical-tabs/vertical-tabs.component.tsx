import { SectionHeader } from '@roche/patterns-indicators/components';
import {
  Container,
  ContentContainer,
  Nav,
  NavTabsList,
  NavTabsListItem,
  NavTabsListItemText,
  NavTabsListTitle,
  SelectedLine,
} from '@roche/patterns-indicators/domains/alert-settings/components/vertical-tabs/vertical-tabs.style';
import { RenderIf } from '@roche/patterns-indicators/utils/markup/render-if.utils';
import * as React from 'react';
import { withTheme } from 'styled-components';

import { Tab } from '../../alert-settings.types';

interface VerticalTab extends Tab {
  label: JSX.Element;
  toggled: boolean;
  selected: boolean;
}

export interface VerticalTabsProps {
  headerText?: string;
  minHeight?: number;
  navItemLabelSelectedStyle?: string;
  setSelectedTabIndex: (index: number) => void;
  tabs?: VerticalTab[];
  theme: any;
}

interface VerticalTabsState {
  selectedTabIndex: number;
}
class VerticalTabsComponent extends React.Component<
  VerticalTabsProps,
  VerticalTabsState
> {
  constructor(props: VerticalTabsProps) {
    super(props);
    this.state = {
      selectedTabIndex: 0,
    };
  }

  public render() {
    const {
      headerText,
      navItemLabelSelectedStyle = '',
      tabs = [],
      theme,
    } = this.props;
    return (
      <Container>
        <Nav onChange={this.tabsChangeHandler}>
          <RenderIf validate={headerText}>
            <NavTabsListTitle>
              <SectionHeader
                borderColor={theme.colors.silverLight}
                title={headerText}
              />
            </NavTabsListTitle>
          </RenderIf>
          <NavTabsList>
            {tabs.map((tab, index) => {
              return (
                <NavTabsListItem
                  key={`tab ${index}`}
                  onClick={
                    tab.enabled ? this.setSelectedTabIndex(index) : undefined
                  }
                  selected={index === this.state.selectedTabIndex}
                  selectedStyle={navItemLabelSelectedStyle}
                >
                  <SelectedLine show={index === this.state.selectedTabIndex} />
                  <NavTabsListItemText>{tab.label}</NavTabsListItemText>
                </NavTabsListItem>
              );
            })}
          </NavTabsList>
        </Nav>
        <ContentContainer>
          {tabs[this.state.selectedTabIndex].content}
        </ContentContainer>
      </Container>
    );
  }

  private setSelectedTabIndex = (index: number) => () => {
    this.props.setSelectedTabIndex(index);
    this.setState({ selectedTabIndex: index });
  };

  private tabsChangeHandler = (event: React.ChangeEvent<any>) => {
    const tabIndex = event.target.id;
    const isChecked = event.target.checked;
    const elementType = event.target.type;

    if (elementType === 'checkbox') {
      if (!isChecked) {
        this.setSelectedTabIndex(0)();
      } else {
        this.setSelectedTabIndex(Number(tabIndex))();
      }
    }
  };
}

export const VerticalTabs = withTheme(VerticalTabsComponent);
