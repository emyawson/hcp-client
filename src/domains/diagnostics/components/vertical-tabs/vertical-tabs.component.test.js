import { shallow } from 'enzyme';
import React from 'react';

import { VerticalTabs } from './vertical-tabs.component';
import { NavTabsListItem } from './vertical-tabs.style';

const mockProps = {
  tabs: [{ label: <span />, keyText: 'test123' }],
};

describe('Vertical Tabs component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<VerticalTabs {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should select a tab onclick', () => {
    const wrapper = shallow(<VerticalTabs {...mockProps} />);
    wrapper
      .find(NavTabsListItem)
      .first()
      .simulate('click');
    expect(wrapper.state().selectedTabIndex).toBe(0);
  });
});
