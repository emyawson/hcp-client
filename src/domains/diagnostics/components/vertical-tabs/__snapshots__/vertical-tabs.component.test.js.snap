// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical Tabs component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VerticalTabs
    tabs={
      Array [
        Object {
          "keyText": "test123",
          "label": <span />,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.nav>
          <styled.ul>
            <NavTabsListItem
              onClick={[Function]}
              selected={true}
              selectedStyle={undefined}
            >
              <styled.div
                show={true}
              />
              <styled.div
                selected={true}
              >
                <span />
              </styled.div>
            </NavTabsListItem>
          </styled.ul>
        </styled.nav>,
        <styled.section
          height={undefined}
        />,
      ],
      "m": undefined,
      "minHeight": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <styled.ul>
            <NavTabsListItem
              onClick={[Function]}
              selected={true}
              selectedStyle={undefined}
            >
              <styled.div
                show={true}
              />
              <styled.div
                selected={true}
              >
                <span />
              </styled.div>
            </NavTabsListItem>
          </styled.ul>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavTabsListItem
                onClick={[Function]}
                selected={true}
                selectedStyle={undefined}
              >
                <styled.div
                  show={true}
                />
                <styled.div
                  selected={true}
                >
                  <span />
                </styled.div>
              </NavTabsListItem>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "test123",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.div
                    show={true}
                  />,
                  <styled.div
                    selected={true}
                  >
                    <span />
                  </styled.div>,
                ],
                "onClick": [Function],
                "selected": true,
                "selectedStyle": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "show": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": <span />,
                    "selected": true,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "height": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.nav>
            <styled.ul>
              <NavTabsListItem
                onClick={[Function]}
                selected={true}
                selectedStyle={undefined}
              >
                <styled.div
                  show={true}
                />
                <styled.div
                  selected={true}
                >
                  <span />
                </styled.div>
              </NavTabsListItem>
            </styled.ul>
          </styled.nav>,
          <styled.section
            height={undefined}
          />,
        ],
        "m": undefined,
        "minHeight": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <styled.ul>
              <NavTabsListItem
                onClick={[Function]}
                selected={true}
                selectedStyle={undefined}
              >
                <styled.div
                  show={true}
                />
                <styled.div
                  selected={true}
                >
                  <span />
                </styled.div>
              </NavTabsListItem>
            </styled.ul>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <NavTabsListItem
                  onClick={[Function]}
                  selected={true}
                  selectedStyle={undefined}
                >
                  <styled.div
                    show={true}
                  />
                  <styled.div
                    selected={true}
                  >
                    <span />
                  </styled.div>
                </NavTabsListItem>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "test123",
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.div
                      show={true}
                    />,
                    <styled.div
                      selected={true}
                    >
                      <span />
                    </styled.div>,
                  ],
                  "onClick": [Function],
                  "selected": true,
                  "selectedStyle": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "show": true,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": <span />,
                      "selected": true,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "span",
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "height": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
